cmd/check.go

```diff
@@ -0,0 +1,117 @@
+// Copyright © 2017 Mesosphere Inc. <http://mesosphere.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"math"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/dcos/3dt/runner"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+const (
+	checkTypeCluster       = "cluster"
+	checkTypeNodePreStart  = "node-prestart"
+	checkTypeNodePostStart = "node-poststart"
+)
+
+var (
+	list          bool
+	checksCfgFile string
+)
+
+// checkCmd represents the check command
+var checkCmd = &cobra.Command{
+	Use:   "check <check-type>",
+	Short: "Execute a DC/OS check",
+	Long: `A DC/OS check can be one of the following types: cluster, node-prestart, node-poststart
+
+cluster ...
+node-prestart ...
+node-poststart ...
+
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		var checkNames []string
+		if len(args) == 0 {
+			cmd.Usage()
+			return
+		} else if len(args) > 1 {
+			checkNames = args[1:]
+		}
+
+		r := runner.NewRunner(defaultConfig.FlagRole)
+		if err := r.LoadFromFile(checksCfgFile); err != nil {
+			logrus.Fatal(err)
+		}
+
+		var (
+			rs  []runner.Response
+			err error
+		)
+
+		switch args[0] {
+		case checkTypeCluster:
+			rs, err = r.Cluster(nil, list, checkNames...)
+		case checkTypeNodePreStart:
+			rs, err = r.PreStart(nil, list, checkNames...)
+		case checkTypeNodePostStart:
+			rs, err = r.PostStart(nil, list, checkNames...)
+		default:
+			logrus.Fatalf("invalid check type %s", args[0])
+		}
+
+		if err != nil {
+			logrus.Fatalf("unable to execute prestart runner: %s", err)
+		}
+
+		emitOutput(rs, list, defaultConfig.FlagVerbose)
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(checkCmd)
+	checkCmd.PersistentFlags().BoolVar(&list, "list", false, "List runner")
+	checkCmd.PersistentFlags().StringVar(&checksCfgFile, "check-config", defaultCheckConfig,
+		"Path to dcos-check config file")
+}
+
+func emitOutput(rs []runner.Response, printOnly, verbose bool) {
```

Comment:
I think `printOnly` here is a better name than `list` that is used above and elsewhere. `printOnly` is much more understandable to the reader.